libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -87,6 +88,44 @@ void StructuralSimplifier::simplify(std::vector<yul::Statement>& _statements)
 			if (expressionAlwaysFalse(*_forLoop.condition))
 				return {std::move(_forLoop.pre.statements)};
 			return {};
+		},
+		[&](ExpressionStatement& _stmt) -> OptionalStatements {
+			FunctionCall& functionCall = get<FunctionCall>(_stmt.expression);
+			if (dynamic_cast<EVMDialect const*>(&m_dialect))
+				if (
+					functionCall.functionName.name == "revert"_yulstring ||
+					functionCall.functionName.name == "return"_yulstring
+				)
```

Comment:
Maybe we can do it for all memory reading opcodes based on the start/length fields in the side-effects...

Reply:
Yep, I think that's better.